content/projects/gsoc/2021/project-ideas/jenkins-custom-distribution-service-updates.adoc

```diff
@@ -0,0 +1,74 @@
+---
+layout: gsocprojectidea
+title: "Custom Jenkins distribution service improvements"
+goal: "Improve the out of the box solution for packaging Jenkins distributions as WAR files or Docker images"
+category: Tools
+year: 2020
+status: published
+skills:
+- Java
+- Web Development
+- Docker
+mentors:
+#To check on the channel post reviews
+#- "martinda"
+#- "sladyn98"
+- "kwhetstone"
+links:
+  gitter: "jenkinsci/platform-sig"
+---
+
+Many users will download Jenkins and spend a great deal of time customizing their environment to be exactly what they need.
+Often this is done through trial and error.
+It would not only be nice to create common configurations to share with team memebers, but with the wider community.
+
+Last summer, a project (link:https://github.com/jenkinsci/custom-distribution-service[custom-distribution-service]) created the base application with a UI and support for uploading configurations.
+There were a lot of suggestions over the course of the summer, to the point where it would make sense to continue the work over another summer.
+The project is currently hibernating though would be useful after issues are addressed.
+
+The service was temporarily offered as a website at \https://customize.jenkins.io.
+More importantly, it should support self-hosting by Jenkins users who want to run it internally. 
+People can select the following configuration items:
+
+* Jenkins core version
+* plugins
+* common configuration, user/password, update-center site e.g.
+* plugin based configuration, Kubernetes, Sonarqube plugin config e.g.
+* backup/restore solution
+* data storage solution
+* multi-form package, jenkins.war or docker image
+* configuration should be scenario-based
+* other things
+
+The entire service exists, but could be greatly expanded on in terms of useability.
+While hosting would be an interesting challenge, initial focus should be on local execution.
+The most important thing to focus on is the backend.
+This needs to be documented, and available over a generic REST api.
+
+There are several extra suggestions defined in the Issues of the main project that can be expanded as the student chooses.
+There are UI enchancements like better notifications of selected plugins, and backend improvements such as REST consistency.
+
```

Comment:
Another interesting area students can explore along with the local execution is the custom configuration of each plugin to be available on the GUI. I am not sure how this would pan out in terms of time since this GSoC is shorter, but it would definitely help